pkg/controller/controller.go

```diff
@@ -184,6 +187,15 @@ func (c *controller) Run(workers int, stopCh <-chan struct{}) {
 		}
 	}
 
+	c.clusterID = string(uuid.NewUUID())
+	func() {
```

Comment:
Seems like you should follow the pattern above and wrap this in a worker, passing the waitGroup and stop channel. It is not clear what the goal of this seamingly random bit of code is doing here in the Run method.

Reply:
There is not any of the features of the worker function. There are none of the args to pass in.

Not sure how to refactor it to look the same such that it can be reused.